adjusted timeouts for inode eviction and garbage collection tests

dashpole opened the pull request:
Inode eviction tests appear to run slower on coreos than the other operating systems I tested on.
I adjusted the timeout for the test from 10 to 30 minutes to compensate.

Garbage collection tests also flake occasionally due to timeouts.
I adjusted the timeout for runtime commands from 2 to 3 minutes, and removed an unused constant.

cc: @Random-Liu

k8s-oncall commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37718)
<!-- Reviewable:end -->

dashpole commented:
@k8s-bot gci gce e2e test this

dashpole commented:
@k8s-bot gce etcd3 e2e test this

dashpole commented:
@Random-Liu This is all good to go!

Random-Liu commented on test/e2e_node/inode_eviction_test.go:
```diff
@@ -206,6 +206,12 @@ func runEvictionTest(f *framework.Framework, testCondition string, podTestSpecs
 		})
 
 		AfterEach(func() {
+			By("deleting pods")
```
We should not delete pods here directly.

We should:
1) check whether disk pressure will be relieved.
2) check whether we can create a new pod.
3) delete all pods in `AfterEach` to make sure everything will be cleaned up no matter the test success or not.

You can reference how disk eviction test is implemented https://github.com/dashpole/kubernetes/blob/31653aff504724438831290677e5bb69effd5b77/test/e2e_node/disk_eviction_test.go.

Random-Liu replied:
If you delete the pod here, we won't know whether it is the eviction relieves the disk pressure, or your deletion.

dashpole commented:
Ok, I moved the "post-eviciton" monitoring code (wait for condition to disappear, making sure condition does not return) to the IT.  The aftereach just deletes the pod.  Note that I added a time.Sleep in the aftereach if the test fails, to make sure that conditions still have a chance to cool down after the test.

rmmh commented:
@k8s-bot gci gce e2e test this (job was accidentally deleted)

dashpole commented:
One bug that I am encountering now is that the "pressure condition" is almost always delayed.

1.  We get Pressure Condition
2. We evict pod 1
3. We evict pod 2
4. Eviction portion of test succeeds
5. Start cooldown portion of test.
5. We get pressure condition again (delayed, but due to pressure from before we evicted pod 2)
6. Test cooldown fails.

I am going to try putting in a delay after the eviction portion of the test push past delayed pressure from the test.

dashpole commented:
Delay appears to solve that issue.  Test ran ~40 times without flaking.  I am having issues pushing, but I will push the change once I resolve my git issues.

dashpole commented:
@k8s-bot verify test this

Random-Liu commented on test/e2e_node/inode_eviction_test.go:
```diff
@@ -203,42 +206,62 @@ func runEvictionTest(f *framework.Framework, testCondition string, podTestSpecs
-		})
 
-		AfterEach(func() {
+			By("waiting for pressure from test to surface")
```
I don't quite understand the comment here.

The pressure should have already surfaced in the previous `Eventually`, right?

Random-Liu commented:
LGTM, please squash your PR.

dashpole committed:
adjusted timeouts; fixed but where pods are deleted late; added delay to account for delayed pressure

dashpole commented:
@Random-Liu Done

Random-Liu commented:
LGTM, thanks for the fix! @dashpole
